test/e2e/network/conntrack.go

```diff
@@ -297,3 +304,22 @@ func dumpConntrack(cs clientset.Interface) {
 		}
 	}
 }
+
+func ipFamilyAddresses(ips []string, family v1.IPFamily) []string {
+	expectedIPs := []string{}
+	if family == v1.IPv4Protocol {
+		for _, ip := range ips {
+			if isIPv4(ip) {
+				expectedIPs = append(expectedIPs, ip)
+			}
+		}
+	}
+	if family == v1.IPv6Protocol {
+		for _, ip := range ips {
+			if netutil.IsIPv6String(ip) {
+				expectedIPs = append(expectedIPs, ip)
+			}
+		}
+	}
```

Comment:
I think this can be simplified to something like
```go
		for _, ip := range ips {
			if netutil.IsIPv6String(ip) == (family == v1.IPv6Protocol) {
				expectedIPs = append(expectedIPs, ip)
			}
		}
```